### DISCUSSION TITLE
Run command on Search

### DISCUSSION THREAD

**Discussion by gryf** [wezterm/wezterm#4254] [Q&A]
Suppose I do have a simple key assignment:

```lua
config.keys = {
  …
  {key = 'u', mods = 'ALT', action = wezterm.action.Search {Regex = 'https?://[\\S]*'}}
}
…
config.key_tables = {
…
 { key = 'o', mods = 'CTRL', action = wezterm.action_callback(
        function(win, pane)
            wezterm.open_with(????, 'xdg-open')
        end)}
}
```

I'd like to use it for opening urls, as search is working on entire scrollback, not only on visible area (like quickselect mode). How can I access currently selected match to be placed in `????` in `wezterm.open_with`?

**Comment by wez** [comment]
Use [QuickSelectArgs](https://wezfurlong.org/wezterm/config/lua/keyassignment/QuickSelectArgs.html) and set `scope_lines` to something big enough to match your scrollback

**Reply by gryf** [reply]
Well, I'm not a big fan of pentadactyl style of selecting links. Is there really no way to move my cursor over matched items and perform whatever action on the selected one?

I'm thinking of something like custom search which would look for sepcific pattern and depending on it, the current matched item will be:
- copied to clipboard/primary/both,
- open in default browser (i.e. using xdg-open)
- open in specific browser (chromium, firefox, or even in the same but different profile)
- execute mpv on youtube/vimeo/etc links
- execute arbitrary in various programs, even depending of the search type - i.e image filepath could be open with a viewer or editor depending on the shortcut
It would be great to even pass a custom key table config (or a function) to such a custom search. Like in the idea below:

```lua

function http_open (win, pane, current)
    local a = wezterm.action
    return { key_table = 
        {
            key = 'c',
            mods = 'CTRL',
            action = a.Multiple { 
                { CopyTo =  'ClipboardAndPrimarySelection' }, 
                { CopyMode =  'Close' } }
        },
        {
            key = 'v',
            mods = 'CTRL',
            action = a.action_callback(wezterm.open_with(current, 'mpv'))
        },
        {
            key = 'Enter',
            mods = 'NONE',
            action = a.action_callback(wezterm.open_with(current, 'xdg-open'))
        }
    }
end


function img_open (win, pane, current)
    local a = wezterm.action
    return { key_table = 
        {
            key = 'c',
            mods = 'CTRL',
            action = a.Multiple { 
                { CopyTo =  'ClipboardAndPrimarySelection' }, 
                { CopyMode =  'Close' } }
        },
        {
            key = 'e',
            mods = 'CTRL',
            action = a.action_callback(wezterm.open_with(current, 'gimp'))
        },
        {
            key = 'Enter',
            mods = 'NONE',
            action = a.action_callback(wezterm.open_with(current, 'gqview'))
        }
    }
end


config.keys = {
    {key = 'u', mods = 'ALT', action = wezterm.action.Search {Regex = 'https?://[\\S]*', func = http_open }},
    {key = 'i', mods = 'ALT', action = wezterm.action.Search {Regex = '/[\\S]*\\.jpg', func = img_open }},
}
```

where there are two different `Search` defined with different set of key_tables, so that matching url will have ability to copy url, play url with mpv or by default (Enter) open in default browser, and the other `Search` will look for absolute filepath ending with `.jpg` and have an option to show it with image viewer or open in editor when invoked with `CTRL-e`.
This is more or less what I want to achieve.
